handler.py

```diff
@@ -14,38 +16,120 @@
-        logger.info("Completed Observatory Scan of " +
+        scanner = PortScanner()
+        results = scanner.scanTCP(target.targetname)
+        logger.info("Completed Port Scan of " +
                     json.dumps(target.targetname))
-        send_to_s3(target.targetname, scan_result)
+        send_to_s3(target.targetname + "_tcpscan", results)
     else:
         logger.error("Target Validation Failed of: " +
                      json.dumps(target.targetname))
 
 
-def runObservatoryScanFromQ(event, context):
+def addObservatoryScanToQueue(event, context):
+    data = json.loads(event['body'])
+    target = Target(data.get('target'))
+
+    if target.isValid():
+        print(SQS_CLIENT.send_message(
+              QueueUrl=os.getenv('SQS_URL'),
+              MessageBody="observatory|" + target.targetname
+              ))
+        return Response({
+            "statusCode": 200,
+            "body": json.dumps({'OK': 'target added to the queue'})
+        }).with_security_headers()
+    
+    else:
+        logger.error("Target Validation Failed of: " +
+                     json.dumps(target.targetname))
+        return Response({
+             "statusCode": 400,
+             "body": json.dumps({'error': 'target was not valid or missing'})
+         }).with_security_headers()
 
-    # This is how we process the hostname
-    # Demonstration purposes, in the final example 
-    # we should extract the hostname and perform a scan
 
+def runScanFromQ(event, context):
+    
+    # Read the queue
     for record, keys in event.items():
         for item in keys:
             if "body" in item:
-                logger.info(item['body'])
+                message = item['body']
+                scan_type, target = message.split('|')
+                if scan_type == "observatory":
+                    scanner = HTTPObservatoryScanner()
+                    scan_result = scanner.scan(target)
+                    send_to_s3(target + "_observatory", scan_result)
+                elif scan_type == "portscan":
+                    scanner = PortScanner()
+                    scan_result = scanner.scanTCP(target)
+                    send_to_s3(target + "_tcpscan", scan_result)
+                else:
+                    # Manually invoked, just log the message
+                    logger.info("Message in queue: " +
+                                json.dumps(message))
+            else:
+                logger.error("Unrecognized message in queue: " +
+                             json.dumps(message))
+
+
+def runScheduledObservatoryScan(event, context):
+
+    # For demo purposes, we will use a static list here
+    target_list = [
+        "www.mozilla.org",
+        "infosec.mozilla.org",
+    ]
+    randomizer = Randomizer(target_list)
+    target = Target(randomizer.next())
+    if target.isValid():
+        logger.info("Tasking Observatory Scan of: " +
+                    json.dumps(target.targetname))
```

Comment:
I'm curious what the purpose of json.dumps'ing the target.targetname if it's just string.  Could  we simply  from the json.dumps part and it would still make sense, or is there specific valid in turning into JSON?

Reply:
I thought of the same thing yesterday, I believe most of this was remains from another iteration of the tools from months ago. I will remove the json parts and see what happens.

Reply:
Removed the json.dumps part from logging messages, no difference. They need to be present for the returned responses from the API endpoints however.